### PR TITLE
Added Mandrill driver

### PR DESCRIPTION
**Pull request by sagikazarmark**

### CODE REVIEW

**Comment by WanWizard** [comment]
@FrenkyNet ?

File: classes/email/driver/mandrill.php [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+
+		foreach ($merge_vars as $key => $value)
+		{
+			$output[] = array(
+				'name' => $key,
+				'content' => $value
+			);
+		}
+
+		if ( ! is_null($rcpt))
```
Comment (frankdejonge) [inline classes/email/driver/mandrill.php]: this check should be done at the start of the function, and returned there.

Comment (sagikazarmark) [inline classes/email/driver/mandrill.php]: Not sure what you mean. The foreach has to run in both cases.

Comment (sagikazarmark) [inline classes/email/driver/mandrill.php]: This whole function will be deleted if you accept my Arr PR in the core. I will replace it with `Arr::keyval_to_assoc` and will do any build process in `_send`.

Comment (sagikazarmark) [inline classes/email/driver/mandrill.php]: Like this
https://github.com/sagikazarmark/email/compare/1.8%2Fdevelop...build_merge_vars

**Comment by sagikazarmark** [comment]
Can I merge the build_merge_vars branch then? Will you merge the core PR?

File: classes/email/driver/mandrill.php (hunk 1 above)
Comment (frankdejonge) [inline classes/email/driver/mandrill.php]: @sagikazarmark correct, overlooked that.

File: classes/email/driver/mandrill.php [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+				'content' => $attachment['contents'],
+			);
+		}
+
+		// Get inline images
+		$images = array();
+
+		foreach ($this->attachments['inline'] as $cid => $attachment)
+		{
+			if (\Str::starts_with($attachment['mime'], 'image/'))
```
Comment (frankdejonge) [inline classes/email/driver/mandrill.php:108]: This restriction should be lifted. That's for the user to decide.

File: classes/email/driver/mandrill.php [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+
+	/**
+	 * Add type to recipient list
+	 *
+	 * @param  string $list to, cc, bcc
+	 * @return array
+	 */
+	protected function build_rcpt($list = 'to')
+	{
+		return array_map(function ($item) use ($list) {
```
Comment (frankdejonge) [inline classes/email/driver/mandrill.php]: The `{` should be on a new line.

**Comment by alrik11es** [comment]
I think that the config should be standardized in a way or another (It's becoming a little bit messy in the event of upcoming more drivers). Not for this pull request I mean in general terms.

**Comment by sagikazarmark** [comment]
I am not sure I understand what you mean, but in v2 Email will probably also change, so config will probably change a lot.

I think with these upcoming services like Mandrill and Mailgun we have to think about the term of "Email abstraction layer", since these services are new layers between sending and receiving emails.

**Comment by alrik11es** [comment]
That is what I was talking about. I see that you have that in your plans, great!

**Comment by sagikazarmark** [comment]
Git rebased, anymore changes needed? Can it be merged?

File: classes/email/driver/mandrill.php [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+		$message_data = \Arr::merge($message_data, $message_options);
+
+		$send_options = extract($this->config['mandrill']['send_options'], EXTR_SKIP);
+
+		$message->send($message_data, $async, $ip_pool, $send_at);
+
+		return true;
+	}
+
+	public function attach($file, $inline = false, $cid = null, $mime = null, $name = null)
```
Comment (frankdejonge) [inline classes/email/driver/mandrill.php:164]: This should have the docblock:

``` php
/**
 * {@inheritdoc}
 */
```

**Pull request merged**